subsys/net/ip/tcp2.c

```diff
@@ -376,10 +376,21 @@ static const char *tcp_conn_state(struct tcp *conn, struct net_pkt *pkt)
 	return buf;
 }
 
-static bool tcp_options_check(void *buf, ssize_t len)
+static u8_t *tcp_options_get(struct net_pkt *pkt, int tcp_options_len)
+{
+	static u8_t options[40]; /* TCP header max options size is 40 */
+
+	net_pkt_skip(pkt, 20); /* skip 20 octets of TCP header */
```

Comment:
preceding th_get already puts the cursor right after the header, so no need to skip anything.

Reply:
For some reason cursor is not after header, and pkt read returns header. The "magic" size will be fixed